File: backend/src/main/java/ca/metricalsky/winston/entity/VideoRestrictionEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "video_restrictions")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VideoRestrictionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "video_id")
    private String videoId;

    @Enumerated(EnumType.STRING)
    @Column(name = "restriction")
    private Restriction restriction;

    @Column(name = "country")
    private String country;

    public enum Restriction {
        ALLOWED,
        BLOCKED,
    }
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/video/VideoRestrictionEntity.kt
package ca.metricalsky.winston.database.entity.video

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.sql.Restriction

@Entity
@Table(name = "video_restrictions")
class VideoRestrictionEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long,

    @Column(name = "video_id")
    var videoId: String?,

    @Enumerated(EnumType.STRING)
    @Column(name = "restriction")
    var restriction: Restriction,

    @Column(name = "country")
    var country: String,
) {
    enum class Restriction {
        ALLOWED,
        BLOCKED
    }
}
